src/Microsoft.Extensions.DependencyModel/DependencyContext.cs

```diff
@@ -38,7 +38,12 @@ public DependencyContext(string target, string runtime, CompilationOptions compi
         private static DependencyContext LoadDefault()
         {
             var entryAssembly = (Assembly)typeof(Assembly).GetTypeInfo().GetDeclaredMethod("GetEntryAssembly").Invoke(null, null);
-            var stream = entryAssembly.GetManifestResourceStream(entryAssembly.GetName().Name + DepsResourceSufix);
+            return Load(entryAssembly);
+        }
+
+        public static DependencyContext Load(Assembly assembly)
+        {
+            var stream = assembly.GetManifestResourceStream(assembly.GetName().Name + DepsResourceSufix);
```

Comment:
If you want to load dependency context for non entry point assembly, for example `xunit` is entry assembly and you wand to get dependency context for test assembly.